envoy/http/codec.h

```diff
@@ -497,6 +498,9 @@ struct Http1Settings {
   // headers set. By default such messages are rejected, but if option is enabled - Envoy will
   // remove Content-Length header and process message.
   bool allow_chunked_length_{false};
+  // Remove HTTP/1.1 Upgrade header tokens matching any provided matcher. By default such
+  // messages are rejected
+  std::shared_ptr<const std::vector<Matchers::StringMatcherPtr>> ignore_upgrade_matchers_;
```

Comment:
Simple `const std::vector<Matchers::StringMatcherPtr>` is enought. We needn't warp it again in the shared pointer.

Reply:
Got it. Seems like a history probolem make use have to accept another problem. But fine.